src/AblyProvider.tsx

```diff
@@ -0,0 +1,80 @@
+import Ably from 'ably';
+import { Types } from 'ably';
+import React from 'react';
+
+const version = '2.1.1';
+
+/**
+ * Wrapper around Ably.Realtime.Promise which injects the 'react-hooks' agent
+ */
+export class Realtime extends Ably.Realtime.Promise {
+    constructor(options: string | Types.ClientOptions) {
+        if (typeof options === 'string') {
+            options = {
+                key: options,
+            } as Types.ClientOptions;
+        }
+
+        (options as any).agents = { 'react-hooks': version };
+
+        super(options);
+    }
+}
+
+interface AblyProviderProps {
+    children?: React.ReactNode | React.ReactNode[] | null;
+    client?: Ably.Types.RealtimePromise;
+    options?: Ably.Types.ClientOptions;
+    id?: string;
+}
+
+type AblyContextType = React.Context<Realtime>;
+
+// An object is appended to `React.createContext` which stores all contexts
+// indexed by id, which is used by useAbly to find the correct context when an
+// id is provided.
+type ContextMap = Record<string, AblyContextType>;
+export const contextKey = '__ABLY_CONTEXT__';
```

Comment:
i did consider this before but i figured it wasn't really worth it since it would break ie11 compatibility and the risk of a naming collision is basically zero anyway. i've now updated it to use `Symbol.for` guarded behind a typeof check